build_tools/github_actions/configure_ci.py

```diff
@@ -0,0 +1,151 @@
+#!/usr/bin/env python3
+
+# Copyright 2022 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+"""Determines whether CI should run on a given PR.
+
+Exit code 0 indicates that it should and exit code 2 indicates that it should
+not.
+"""
+
+import fnmatch
+import os
+import subprocess
+import sys
+from typing import Iterable, Mapping, MutableMapping
+
+SKIP_CI_KEY = "skip-ci"
```

Comment:
Could document this in https://github.com/iree-org/iree/blob/main/CONTRIBUTING.md#presubmits maybe (though changing that file triggers a notification for every developer like "the contribution guidelines for this project changed")

Reply:
Yeah I also wasn't sure how much we want to advertise this :-) Maybe I'll collect a tranche of updates and make them all at once to avoid spamming people.